homeassistant/components/neato/__init__.py

```diff
@@ -77,10 +70,21 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         )
     )
 
+    session = config_entry_oauth2_flow.OAuth2Session(hass, entry, implementation)
+    try:
+        await session.async_ensure_token_valid()
+    except aiohttp.ClientResponseError as ex:
```

Comment:
Don't catch all errors. This means that if there is no internet connection during startup, it will trigger re-auth. 

If you look at Home Assistant, we return a bad request status code if it's invalid:

https://github.com/home-assistant/core/blob/dev/homeassistant/components/auth/__init__.py#L303-L315

I don't know what neato does.

Reply:
We need to check the response status code here and only raise reauth for codes 401 and 403.

Reply:
[Beehive API - User Authentication](https://developers.neatorobotics.com/api/beehive): 401 and 403 ;-)

Code updated.

Simone